Fix GCC 8 compile warning

I noticed a build warning when compiling with GCC 8. Here's a [Travis log](https://travis-ci.com/OpenVisualCloud/SVT-AV1/jobs/239323880#L680).
```
[  0%] Building C object Source/Lib/Common/Codec/CMakeFiles/COMMON_CODEC.dir/EbAdaptiveMotionVectorPrediction.c.o
/home/travis/build/OpenVisualCloud/SVT-AV1/Source/Lib/Common/Codec/EbAdaptiveMotionVectorPrediction.c: In function ‘setup_ref_mv_list’:
/home/travis/build/OpenVisualCloud/SVT-AV1/Source/Lib/Common/Codec/EbAdaptiveMotionVectorPrediction.c:1120:59: error: array subscript 2 is above array bounds of ‘IntMv[2]’ {aka ‘union IntMv[2]’} [-Werror=array-bounds]
                     comp_list[comp_idx][idx] = ref_id[idx][list_idx];
                                                ~~~~~~~~~~~^~~~~~~~~~
cc1: all warnings being treated as errors
```
Blocks #646.

--- a/Source/Lib/Common/Codec/EbAdaptiveMotionVectorPrediction.c
+++ b/Source/Lib/Common/Codec/EbAdaptiveMotionVectorPrediction.c
@@ -1108,23 +1108,21 @@ void setup_ref_mv_list(
                 idx += mi_size_high[candidate_bsize];
             }
 
-            //CHKN I dont think we need the third
-
             // Build up the compound mv predictor
-            IntMv comp_list[3][2];
+            IntMv comp_list[MAX_MV_REF_CANDIDATES][2];
 
             for (int32_t idx = 0; idx < 2; ++idx) {
                 int32_t comp_idx = 0;
-                for (int32_t list_idx = 0; list_idx < ref_id_count[idx]
-                    && comp_idx < 3; ++list_idx, ++comp_idx)
-                    comp_list[comp_idx][idx] = ref_id[idx][list_idx];
-
-                for (int32_t list_idx = 0; list_idx < ref_diff_count[idx]
-                    && comp_idx < 3; ++list_idx, ++comp_idx)
-                    comp_list[comp_idx][idx] = ref_diff[idx][list_idx];
-
-                for (; comp_idx < 3; ++comp_idx)
-                    comp_list[comp_idx][idx] = gm_mv_candidates[idx];
+                for (int32_t list_idx = 0;
+    	             list_idx < ref_id_count[idx] && comp_idx < MAX_MV_REF_CANDIDATES;
+	             ++list_idx, ++comp_idx)
+                  comp_list[comp_idx][idx] = ref_id[idx][list_idx];
+                for (int32_t list_idx = 0;
+    	            list_idx < ref_diff_count[idx] && comp_idx < MAX_MV_REF_CANDIDATES;
+	            ++list_idx, ++comp_idx)
+                  comp_list[comp_idx][idx] = ref_diff[idx][list_idx];
+                for (; comp_idx < MAX_MV_REF_CANDIDATES; ++comp_idx)
+                  comp_list[comp_idx][idx] = gm_mv_candidates[idx];
             }
 
             //CHKN fill the stack, increment the counter

